src/ui/ui_settings/ui_settings_service_factory.js

```diff
@@ -0,0 +1,25 @@
+import { UiSettingsService } from './ui_settings_service';
+
+/**
+ *  Create an instance of UiSettingsService that will use the
+ *  passed `callCluster` function to communicate with elasticsearch
+ *
+ *  @param  {Function} callCluster should accept a method name as a string and a params object
+ *  @return {UiSettingsService}
+ */
+export function uiSettingsServiceFactory(server, options = {}) {
```

Comment:
> As a side note, I usually try to avoid property-bag-as-the-only-function-argument. 

Yeah, I generally do my best to keep function arguments to a one/two values, but when there are more, or some/all of them are optional, I move to "named args" for a couple reasons:

1. doesn't lead to multi-line argument lists
1. makes it easier to specify a couple args and not all of them
1. makes it really easy to merge arguments/provide defaults in wrappers
1. I find it really easy to read when the options value is destructured as the first statement in the code